src/Tabs/TabsWrapper.php

```diff
@@ -132,9 +131,25 @@ function twSetup(tabsid) {
   twObject[tabsid].counter = 100;
   // Close icon: removing the tab on click
   nav.on("click", "span.icon-close", function() {
-    var panelId = $(this).parent().attr("href").substring(1);
-    top.restoreSession();
-    twCloseTab(tabsid, panelId);
+    const self = $(this);
+    const closeTab = function() {
+        const panelId = self.parent().attr("href").substring(1);
+        top.restoreSession();
+        twCloseTab(tabsid, panelId);
+        localStorage.removeItem('isSoapEdit');
```

Comment:
shouldn't need to use localStorage. would be more straightforward to just store it in top.* . See here for example of flag that is stored there:
https://github.com/openemr/openemr/blob/master/interface/main/tabs/main.php#L157-L164

Reply:
Wow, Thanks for your instruction, It's very useful to me,
I will try it! 😁